server/svix-server/src/queue/redis.rs

```diff
@@ -51,9 +55,13 @@ pub async fn new_pair(pool: RedisPool) -> (TaskQueueProducer, TaskQueueConsumer)
         let batch_size: isize = 50;
         let task_validity_duration = chrono::Duration::from_std(TASK_VALIDITY_DURATION).unwrap();
 
-        let pool = worker_pool;
+        let mut pool = worker_pool.get().await.unwrap();
+
+        // drain legacy queues:
+        migrate_legacy_queues(&mut pool).await;
```

Comment:
This never frees the pool connection, should be:
```suggestion
        let pool = worker_pool;
        {
          let mut pool = worker_pool.get().await.unwrap();

          // drain legacy queues:
          migrate_legacy_queues(&mut pool).await;
       }
```